### PR TITLE
extract runtime abstractions into own crate

### PR DESCRIPTION
**Pull request by alexytsu**
Small prefactor to prepare for more work on the NFT library

### CODE REVIEW

File: fvm_actor_utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "fvm_actor_utils"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (abright) [inline fvm_actor_utils/Cargo.toml:6]: Don't need this placeholder comment now

Might also be good to fill in metadata here so it's ready for publishing

File: fvm_actor_utils/src/messaging.rs [hunk 2]
```diff
@@ -52,8 +52,8 @@ pub trait Messaging {
     fn initialize_account(&self, address: &Address) -> Result<ActorID>;
 }
 
-// the method number comes from taking the name as "Received" and applying
-// the transformation described in https://github.com/filecoin-project/FIPs/pull/399
+// the method number comes from taking the name as "Receive" and applying
+// the transformation described in [FRC-0042](https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0042.md)
```
Comment (abright) [inline fvm_actor_utils/src/messaging.rs:56]: Should these be doc comments if using markdown syntax?

**Review by abright — APPROVED** [review]

File: frc46_token/src/receiver/mod.rs [hunk 3]
```diff
@@ -1,9 +1,9 @@
+use fvm_actor_utils::messaging::{Messaging, RECEIVER_HOOK_METHOD_NUM};
```
[the remark this answers is not in the document]
Comment (alexytsu) [reply frc46_token/src/receiver/mod.rs:1]: Will do this once the universal receiver hook is in its own crate

File: fvm_actor_utils/Cargo.toml (hunk 1 above)
Comment (alexytsu) [reply fvm_actor_utils/Cargo.toml:6]: Done in followup https://github.com/helix-onchain/filecoin/pull/108

File: fvm_actor_utils/src/messaging.rs (hunk 2 above)
Comment (alexytsu) [reply fvm_actor_utils/src/messaging.rs:56]: Done in followup https://github.com/helix-onchain/filecoin/pull/108

**Pull request merged**